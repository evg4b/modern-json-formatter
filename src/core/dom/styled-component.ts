import { registerStyle } from '@core/ui/helpers';

export abstract class StyledComponentElement extends HTMLElement {
  protected readonly shadow = this.attachShadow({ mode: 'closed' });
  protected readonly _styles: HTMLStyleElement;

  protected constructor(styles: string) {
    super();
    this._styles = registerStyle(this.shadow, styles);
  }
}
