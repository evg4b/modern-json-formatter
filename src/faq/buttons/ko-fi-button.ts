import { BaseButtonElement } from './button';
import icon from './ko-fi-button-icon.svg';

class KoFiButton extends BaseButtonElement {
  constructor() {
    super(icon);
  }
}

customElements.define('ko-fi-button', KoFiButton);
