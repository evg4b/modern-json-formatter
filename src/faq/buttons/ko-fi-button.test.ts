import '@testing/svg.mock';
import { describe, expect, it } from '@jest/globals';
import { getShadowRoot } from '@testing/styled-component';
import { KoFiButton } from './ko-fi-button';

describe('ko-fi-button', () => {
  it('should render button link', () => {
    const button = new KoFiButton();
    button.attributeChangedCallback('href', '', 'https://ko-fo.com');
    button.attributeChangedCallback('title', '', 'Support the project');
    const root = getShadowRoot(button);
    expect(root.innerHTML).toMatchSnapshot();
  });
});
