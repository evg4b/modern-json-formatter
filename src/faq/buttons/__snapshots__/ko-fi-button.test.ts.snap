// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ko-fi-button should render button link 1`] = `
"<style type="text/css" rel="stylesheet" role="presentation">
:host {
  display: flex;
  flex-direction: column;

  a {
    display: flex;
    flex-direction: column;
    transition: all 0.2s ease-in-out;

    &:hover {
      transform: scale(1.2);
    }
  }
}
</style><a target="_blank" href="https://ko-fo.com" title="Support the project"><svg>ko-fi-button-icon</svg></a>"
`;
